sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_authentication_async.py

```diff
@@ -4,35 +4,131 @@
 # license information.
 #-------------------------------------------------------------------------
 from functools import partial
+import time
+from typing import Awaitable, Callable, Optional, Tuple, Union, Any
+
+from azure.servicebus._pyamqp.sasl import SASLAnonymousCredential
 
 from ..authentication import (
     _generate_sas_access_token,
-    SASTokenAuth,
-    JWTTokenAuth
+    AccessToken
 )
-from ..constants import AUTH_DEFAULT_EXPIRATION_SECONDS
-
+from ..constants import AUTH_DEFAULT_EXPIRATION_SECONDS, AUTH_TYPE_CBS, TOKEN_TYPE_JWT, TOKEN_TYPE_SASTOKEN
 
-async def _generate_sas_token_async(auth_uri, sas_name, sas_key, expiry_in=AUTH_DEFAULT_EXPIRATION_SECONDS):
+async def _generate_sas_token_async(
+        auth_uri: str,
+        sas_name: str,
+        sas_key: str,
+        expiry_in: float = AUTH_DEFAULT_EXPIRATION_SECONDS
+    ) -> AccessToken:
     return _generate_sas_access_token(auth_uri, sas_name, sas_key, expiry_in=expiry_in)
 
+class _CBSAuthAsync:
+    # TODO:
+    #  1. naming decision, suffix with Auth vs Credential
+    auth_type = AUTH_TYPE_CBS
+
+    def __init__( # pylint: disable=unused-argument
+        self,
+        uri: str,
+        audience: str,
+        token_type: Union[str, bytes],
+        get_token: Callable[[], Awaitable[AccessToken]],
+        *,
+        expires_in: float = AUTH_DEFAULT_EXPIRATION_SECONDS,
```

Comment:
would this be Optional too? in the method below, it says must provide this or expires_on.